lib/plug/csrf_protection.ex

```diff
@@ -103,17 +103,34 @@ defmodule Plug.CSRFProtection do
 
   def init(opts), do: opts
 
-  def call(conn, _opts) do
+  def call(conn, opts) do
+    opts = verify_opts opts
     csrf_token = get_session(conn, "_csrf_token")
     Process.put(:plug_csrf_token, csrf_token)
-
     if not verified_request?(conn, csrf_token) do
-      raise InvalidCSRFTokenError
+      if opts[:with] == :exception do
```

Comment:
If/when `CSRFProtection`'s options become more complicated, then we can change the API to make it as complex as it needs to be. We may not even reach that point.